projects/ADuCM360_demo_cn0411/src/CN0411.c

```diff
@@ -0,0 +1,1876 @@
+ * @param output_val - voltage value to be written on DAC
+ * @return 0 in case of success, negative error code otherwise.
+*/
+int32_t CN0411_DAC_set_value(cn0411_device *cn0411_dev, float output_val)
+{
+	int32_t ret;
+	uint32_t dac_val;
+
+	dac_val = (uint32_t)(output_val * DAC_FS_VAL / VREF);
+	cn0411_dev->ad5683_dev.dac_reg_value = dac_val;
+	ret = AD5683_write_dac_value(&cn0411_dev->ad5683_dev,
+				     cn0411_dev->ad5683_dev.dac_reg_value);
+
+	return ret;
+}
+
+/**
+ * CN0411 ADC Operation Mode function
+ *
+ * @param cn0411_dev - The device structure.
+ * @param mode - operation mode of the ADC
+ * @return 0 in case of success, negative error code otherwise.
+*/
+int32_t CN0411_ADC_operation_mode (cn0411_device * cn0411_dev, opMode mode)
+{
+	int32_t ret;
+
+	cn0411_dev->ad7124_dev.regs[AD7124_ADC_Control].value =
+		(AD7124_ADC_CTRL_REG_REF_EN
+		 | AD7124_ADC_CTRL_REG_DATA_STATUS
+		 | AD7124_ADC_CTRL_REG_POWER_MODE(LOW_POWER)
+		 | AD7124_ADC_CTRL_REG_MODE(mode)
+		 | AD7124_ADC_CTRL_REG_CLK_SEL(INTERNAL_CLK1));
+
+	ret = AD7124_WriteRegister(&cn0411_dev->ad7124_dev,
+				   cn0411_dev->ad7124_dev.regs[AD7124_ADC_Control]);
+
+	return ret;
+}
+
+/**
+ * CN0411 ADC Setup 0 configuration function
+ *
+ * @param cn0411_dev - The device structure.
+ * @return 0 in case of success, negative error code otherwise.
+*/
+int32_t CN0411_ADC_setup0 (cn0411_device *cn0411_dev)
+{
+	int32_t ret;
+
+	cn0411_dev->ad7124_dev.regs[AD7124_Config_0].value =
+		(AD7124_CFG_REG_BURNOUT(0) | AD7124_CFG_REG_REF_BUFP
+		 | AD7124_CFG_REG_REF_BUFM | AD7124_CFG_REG_AIN_BUFP
+		 | AD7124_CFG_REG_AINN_BUFM | AD7124_CFG_REG_REF_SEL(0)
+		 | AD7124_CFG_REG_PGA(0));
+
+	ret = AD7124_WriteRegister(&cn0411_dev->ad7124_dev,
+				   cn0411_dev->ad7124_dev.regs[AD7124_Config_0]);
+
+	return ret;
+}
+
+/**
+ * CN0411 ADC Setup 1 configuration function
+ *
+ * @param cn0411_dev - The device structure.
+ * @return 0 in case of success, negative error code otherwise.
+*/
+int32_t CN0411_ADC_setup1 (cn0411_device *cn0411_dev)
+{
+	int32_t ret;
+
+	cn0411_dev->ad7124_dev.regs[AD7124_Config_1].value =
+		(AD7124_CFG_REG_BURNOUT(0)
+		 | AD7124_CFG_REG_REF_BUFP
+		 | AD7124_CFG_REG_REF_BUFM
+		 | AD7124_CFG_REG_AIN_BUFP
+		 | AD7124_CFG_REG_AINN_BUFM
+		 | AD7124_CFG_REG_REF_SEL(2)
+		 | AD7124_CFG_REG_PGA(0));
+	ret = AD7124_WriteRegister(&cn0411_dev->ad7124_dev,
+				   cn0411_dev->ad7124_dev.regs[AD7124_Config_1]);
+
+	return ret;
+}
+
+/**
+ * CN0411 ADC Channel 0 configuration function
+ *
+ * @param cn0411_dev - The device structure.
+ * @param ch_en - enable/disable ADC channel
+ * @return 0 in case of success, negative error code otherwise.
+*/
+int32_t CN0411_ADC_ch0 (cn0411_device *cn0411_dev, uint8_t ch_en)
+{
+	int32_t ret;
+
+	cn0411_dev->ad7124_dev.regs[AD7124_Channel_0].value = ((ch_en << 15)
+			| AD7124_CH_MAP_REG_SETUP(0)
+			| AD7124_CH_MAP_REG_AINP(1)
+			| AD7124_CH_MAP_REG_AINM(6));
+	ret = AD7124_WriteRegister(&cn0411_dev->ad7124_dev,
```

Comment:
return directly and leave a blank line before; check this everywhere in your code.